### DISCUSSION TITLE
Clarification of options (Google Takeout)

### DISCUSSION THREAD

**Discussion by MaximumFish** [simulot/immich-go#426] [General]
Hi, 

First of all let me thank you for writing this tool. I'm finally getting around to doing something with my GT images and this looks like the perfect solution! I just wanted some clarification of some of the options before I run it in anger...

Specifically when using the `-google-photos` option:

- `stack-burst` doesn't behave differently for takeouts, presuambly, so I still need to specify this to have those imported?
- `create-stacks` - the way I'm reading it, the above option will do nothing unless this one is also specified?
- I'm a bit uncertain as to how `keep-untitled-albums` works. 99% of my photos are just an unsorted stream, but I see the Takeout archive sorts them into years, at least in terms of the folder structure. Do those count as albums? I just want to import everything.

I think that's everything I'm unsure about. Thanks again for writing this!

**Comment by simulot** [comment]
When not active, photos are uploaded as they are. 
When active, the related photos are placed in a immich's stack.
There are opened issues for both stacks and burst... So don't expect too much of them. Options are inactive by default.

It's possible to have untitled album in Google Photos, thanks to sharing feature. They are ignored by default. 
The option `keep-untitled-albums` just create same untitled albums in immich.

**Reply by MaximumFish** [reply]
Sorry I'm still none-the-wiser.

- If I want burst stacks I should run it with _both_ `create-stacks` and `stack-burst`? (but it might not work very well due to unresolved issues)
- If untitled albums are ignored, does that mean it skips all photos in those albums or just that it doesn't sort them into various albums, presumably called "untitled"? I do have shared photos from both my wife and another of my own Google accounts from when I did a migration.

Like I said, I don't want to skip anything, but wondering what happens to photos that I haven't _specifically/manually put into an album_.

Thanks again!

**Reply by simulot** [reply]
The safe is to import bursts  without the options. 
I'll work on a dedicated tool for stacking bursts already uploaded 

Photos belonging to an untitled album are uploaded, without creating an album. With the option `keep-untitled-albums`, they will be placed in an album. 

Friends shared photos aren't present in the takeout. This is a google limitation. The only exception is the partner's share.

**Reply by MaximumFish** [reply]
Great, thanks, that helps. :)
